### PR TITLE
feat(cloudwatch): add verticalAnnotations property to GraphWidget

### PR DESCRIPTION
**Pull request by brendo-m**
Adds a `verticalAnnotation` property to `GraphWidget`. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html#CloudWatch-Dashboard-Properties-Annotation-Format.

Also see https://github.com/aws/aws-cdk/issues/7622

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by brendo-m** [comment]
Just realized there are breaking changes here - interface `Shading` has been renamed to `HorizontalShading` and there's a new `VerticalShading`.

Should I revert `HorizontalShading` to `Shading` and keep `VerticalShading`? It's a little unclear but not the end of the world

**Comment by brendo-m** [comment]
I just went ahead and made that change

**Comment by gshpychka** [comment]
@RomainMuller any change of this to get merged?

File: packages/@aws-cdk/aws-cloudwatch/lib/graph.ts [hunk 1]
```diff
@@ -281,6 +288,11 @@ export class GraphWidget extends ConcreteWidget {
       ...(this.props.leftAnnotations || []).map(mapAnnotation('left')),
       ...(this.props.rightAnnotations || []).map(mapAnnotation('right')),
     ];
+    const verticalAnnotations = this.props.verticalAnnotations;
+    const annotations = horizontalAnnotations.length > 0 || verticalAnnotations ? ({
+      horizontal: horizontalAnnotations.length > 0 ? horizontalAnnotations : undefined,
+      vertical: verticalAnnotations,
+    }) : undefined;
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-cloudwatch/lib/graph.ts:295]: Should we check for `verticalAnnotations.length > 0` ?

**Review by madeline-k — CHANGES_REQUESTED** [review]
Thank you for your contribution, @brendo-m! We just need a description and an example added to the README before this can be merged.

**Review by madeline-k — CHANGES_REQUESTED** [review]
We need an update to the README with a description and example for this feature.

**Comment by TheRealAmazonKendra** [comment]
This PR has been deemed to be abandoned, and will be closed. Please create a new PR for these changes if you think this decision has been made in error.

**Comment by brendo-m** [comment]
Reviving this PR in https://github.com/aws/aws-cdk/pull/22568

**Pull request closed without merging**